packages/elements/src/components-v3/Card/card-components.tsx

```diff
@@ -0,0 +1,220 @@
+import { Intent } from '../../helpers/v3/intent'
+import { useMediaQuery } from '../../hooks/use-media-query'
+
+export interface CardListItem {
+  listCardItemHeading?: string
+  listCardItemSubHeading?: string
+  listCardItemIcon?: IconNames
+  onClick?: () => void
+}
+
+export interface ContextMenuItem {
+  icon: IconNames
+  onClick: () => void
+  intent?: Intent
+}
+
+export interface CardContextMenuProps extends HTMLAttributes<HTMLDivElement> {
+  contextMenuItems?: ContextMenuItem[]
+}
+
+export interface CardProps extends HTMLAttributes<HTMLDivElement> {
+  isMainCard?: boolean
+  mainCardHeading?: string
+  mainCardSubHeading?: string
+  mainCardSubHeadingAdditional?: string
+  mainCardBody?: string
+  mainCardImgUrl?: string
+  mainContextMenuItems?: ContextMenuItem[]
+  isListCard?: boolean
+  listCardHeading?: string
+  listCardSubHeading?: string
+  listCardItems?: CardListItem[]
+  listContextMenuItems?: ContextMenuItem[]
+  isSelected?: boolean
+}
+
+export const handleToggleContextMenu = (
+  contextMenuOpen: boolean,
+  setContextMenuOpen: Dispatch<SetStateAction<boolean>>,
+) => () => {
+  setContextMenuOpen(!contextMenuOpen)
+}
+
+export const handleToggleMainMobileOpen = (
+  mainMobileOpen: boolean,
+  setMainMobileOpen: Dispatch<SetStateAction<boolean>>,
+) => () => {
+  setMainMobileOpen(!mainMobileOpen)
+}
+
+export const handleToggleListMobileOpen = (
+  listMobileOpen: boolean,
+  setListMobileOpen: Dispatch<SetStateAction<boolean>>,
+) => () => {
+  setListMobileOpen(!listMobileOpen)
+}
+
+export const handleToggleBothMobileOpen = (
+  mainMobileOpen: boolean,
+  setMainMobileOpen: Dispatch<SetStateAction<boolean>>,
+  listMobileOpen: boolean,
+  setListMobileOpen: Dispatch<SetStateAction<boolean>>,
+) => () => {
+  setMainMobileOpen(!mainMobileOpen)
+  setListMobileOpen(!listMobileOpen)
+}
+
+export const CardContextMenu: FC<CardContextMenuProps> = ({ className, contextMenuItems, ...rest }) => {
+  const [contextMenuOpen, setContextMenuOpen] = useState<boolean>(false)
+  if (!contextMenuItems) return null
+  return (
+    <CardContextMenuWrapper className={className} {...rest}>
+      <CardContextMenuToggle onClick={handleToggleContextMenu(contextMenuOpen, setContextMenuOpen)}>
+        <Icon icon="more" fontSize="1.25rem" />
+      </CardContextMenuToggle>
+      <CardContextMenuItems className={cx(contextMenuOpen && elCardContextMenuOpen)}>
+        <CardContextMenuItem onClick={handleToggleContextMenu(contextMenuOpen, setContextMenuOpen)}>
+          <Icon icon="close" />
+        </CardContextMenuItem>
+        {contextMenuItems.map(({ icon, intent, onClick }, index) => (
+          <CardContextMenuItem key={index} onClick={onClick}>
+            <Icon icon={icon} intent={intent} />
+          </CardContextMenuItem>
+        ))}
+      </CardContextMenuItems>
+    </CardContextMenuWrapper>
+  )
+}
+
+export const Card: FC<CardProps> = ({
+  className,
+  isMainCard,
+  isListCard,
```

Comment:
should this be changed to `cardType: 'isMainCard' | 'isListCard'`? As the current condition relies on one of these being set. I'm finding it a little confusing to follow? In the tests I can see both being used? But in the component if neither is set then no children are visable. Is this correct?

Reply:
Yes, it can be either or both - worth looking at the designs :-)

Reply:
thanks!